### PR TITLE
Add community expectations about conduct and reviewing.

### PR DESCRIPTION
**Pull request by brendandburns**
@sarahnovotny @smarterclayton @philips @thockin @davidopp @jbeda 

As discussed in the previous community meeting here's a description of the expectations around community, I expanded it a little bit beyond code review because it seemed like a good place to put it
down in one location.

Please let me know what I got wrong (or right ;)

Thanks
--brendan

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32096)

### CODE REVIEW

File: community-expectations.md
Comment (thockin) [inline community-expectations.md]: This is one heck of a sentence. The first part is missing a verb - probably 'is' ?

I would break and start a new sentence after "distributed systems".  "...distributed systems.  A second, equally important goal..."

misspelled orchestration

File: community-expectations.md
Comment (thockin) [inline community-expectations.md]: s/or/are/

File: community-expectations.md
Comment (thockin) [inline community-expectations.md]: I don't really like quantifying like this.  I am also worried about declaring "expectations" of people.  It feels condescending, somehow. I totally ACK the need to get more reviews, but the bar you have set is pretty low to have expectations...

File: community-expectations.md
Comment (thockin) [inline community-expectations.md]: Again this just "feels" bad.  It feels petty.  I know we need something to motivate a broader base, but codifying it this way feels wrong.

File: community-expectations.md
Comment (davidopp) [inline community-expectations.md]: +1, these numbers seem really arbitrary and could have the unintended consequence of causing people to hold back activity to avoid reaching it, if they really want to avoid reviewing.

I'd rather see expectation to do code reviews built into a more holsitic contributors ladder.

File: community-expectations.md
Comment (davidopp) [inline community-expectations.md]: +1, and what does "will also suffer increased latency" mean? Like we will intentionally delay reviews for people who aren't reviewing? How would you enforce that?

**Comment by davidopp** [comment]
Can this be combined this with one of the existing files in docs/devel? We have so many similar documents there... collab.md seems like it might be a good place for this.

**Comment by davidopp** [comment]
(By similar I mean in the spirit of expectations for contributing to the project.)

File: community-expectations.md
Comment (eparis) [inline community-expectations.md]: s/orcheastration/orchestration/

File: community-expectations.md
Comment (smarterclayton) [inline community-expectations.md]: remove `, as such, it `

File: community-expectations.md
Comment (smarterclayton) [inline community-expectations.md]: I think it's critically important that the first sentence here needs to be a royal we - i.e. "We the community therefore describe our expectations for our members."   It can't be dispassionate outsider.

**Comment by krmayankk** [comment]
great start with good intentions.  its interesting how we will enforce any of the things mentioned above. We will always have some people highly motivated to do both reviews and PR, some who will be just for one or the other. There will be some people who just need one feature, and they will follow up until that feature they want is merged or issue fixed and in the process will develop deep expertise around that feature and once that is done, they have other critical issues on their plate from the organization they represent, that they will move on. They will be active contributors during that period.

File: community-expectations.md
Comment (roberthbailey) [inline community-expectations.md]: Any chance we can have the reviewer assignment honor an OOO config somewhere? I've probably been ignoring PRs for the last month while on leave.....

**Comment by brendandburns** [comment]
@thockin @davidopp I think that is super important that it is clear that these are expectations, it's not optional.

And in particular, we need hard numbers because Brandon is going to run a script and we're going to automatically opt-in people to reviewing who meet this criteria.

This is a big part of why I brought this up in community meeting, we need to decide (as a community) if we are going to require people who are active in the community to do code review.  It seems to me that we can either maintain the status quo where fewer people review than contribute and therefore review latency and load is high, or we can go to something like this, where active participation in the community means active participation in all parts of the community.

By all means we can try an "all carrots" approach whereby we don't penalize people who don't do reviews but we do reward people who do a lot of reviews (not exactly sure what that looks like, but I'm sure it can be done)

But I think we are going to have to quantify what "active" means, if only so that we can write the code that opts people in to being reviewers.

Paging @sarahnovotny for her thoughts on this.

File: community-expectations.md
Comment (brendandburns) [inline community-expectations.md]: Yes, this has been requested elsewhere and will definitely be part of the system.

**Comment by philips** [comment]
For everyone's context this conversation was started when we discussed merging a list of reviewers into the OWNERS files (details on https://github.com/kubernetes/kubernetes/pull/31567). The motivation for introducing reviewers is to increase velocity by sharding the toil of code review across more people.

However, when we put this PR forward to add reviewers two things happened:

1. People wanted to move to an opt-in vs. opt-out model for reviewers
2. People debated the improvement that the reviewers system will deliver without a carrot

In the Kubernetes community I have very often heard the refrain "if people review more velocity will increase and your stuff will get merged faster". Now, I think there is some glimpse of truth to that but I think there are blocks to making that true:

1. Quality reviews by a small group that the owner/assignee trusts
2. Documented community expectation about the job of reviewers
3. A carrot or stick to encourage reviewer-ship an activity folks do

Today, I feel like 1 & 2 are sufficient to get things going. And I think we should hold off on 3 until we see how well or poorly 1&2 work.

Given all that, I think we should:

- Merge this PR minus carrot/sticks so the people doing the work to automate the collection of reviewers in https://github.com/kubernetes/kubernetes/pull/31567 and the people being volunteered to review know the why/how.
- Hold off on the carrot/stick conversation until we see how this first round of getting people to do reviews go. If we later feel we need a stick/carrot we can add one.

**Comment by brendandburns** [comment]
@philips thanks for your thoughts.

I'm fine w/ merging this minus the carrots/sticks and see how we do.

@thockin @davidopp what do you think?  If you agree, I'll revise this PR to remove carrots/sticks but retain the numbers to quantify what "active" means.

**Comment by sarahnovotny** [comment]
I like the definitions of what active is.  If we have metrics, we can follow with the more subjective lense "are these people contributing in different but important ways that are ok or better".  

I see this as doing 2 things 

1. encouraging more reviews by *everyone* to grow the reviewer base, and decrease the load on the maintainers. ideally, though probably eventually, increasing the velocity of the review/merge process

but, more importantly

2. to showcase the skills of the new/different contributors as a means to establish explicitly referenceable trust and eventually grow the maintainer ranks.

@davidopp has a good point that we need to consolidate and rationalize the many places we're collecting what contributing means type of docs... I have talked to both the Intel and CoreOS lots (via @ConnorDoyle and @philips) about having their next newbies help with that as they join the project and learn the cultural expectations.

**Comment by brendandburns** [comment]
Ping to @thockin and @davidopp

File: community-expectations.md
Comment (smarterclayton) [inline community-expectations.md]: IF we're going to set exact numbers, we should double the margin in all directions.  Usually "a bar" is all that is required - if you said "1 PR in the last two months".

After the list, please describe the "why" of the list:

These criteria are intended to help us automate processes that deal with the large number of contributors to the project and helps us manage the project.  They are not intended to penalize members of the community with occasional contributions or who contribute in other spaces.

I.e. a bit more warm fuzzy.

File: community-expectations.md
Comment (smarterclayton) [inline community-expectations.md]: Agree we need to reword this in a slightly different way.  We should probably say that we want to properly incentivize reviewers and reward them, but that the mechanism for encouraging that is still under discussion.

File: community-expectations.md
Comment (thockin) [inline community-expectations.md]: Agree with Clayton, that bar is too low.

Also, are we committing to tracking this?  Who is going to write the bot that adds and removes people from the reviewer pool?

I'm really uncomfortable with the numbers.  I am worried that, as soon as we start assigning reviews, this could be chilling.  Especially if the bar is low.  Someone who fixes typos, adds comments, clarifies errors, etc could EASILY hit 2 or 4 PRs in a month.  Someone who does some testing for their deployment could easily file 3 or more issues.  Commenting on PRs is cheap - do we want to discourage that?

This still feels like negative reinforcement to me, and we all know that doesn't work long term.  "If you want to play you have to review".  Can't we twist it around and just get the ladder defined - if you do X you are a "star reviewer"r.  If you do Y (> X), you are a "maintainer".  If you do Z (> Y) you are an "owner".  Positive reinforcement.

**Comment by thockin** [comment]
There are a lot of comments still open on the doc, but my strongest feeling has been restated.  I'll defer to @sarahnovotny . If you say we need this, I'll follow your lead, but all my "training" as a parent is kicking in saying this is structured backwards.

**Comment by ConnorDoyle** [comment]
Just have 2c to drop into the conversation: all reviews not equal. e.g. "carrot" motivated reviews are likely to be of much higher quality than "stick" motivated reviews. So although either method might yield the same number the qualitative part, which the bot cannot measure easily, could vary a lot.

**Comment by philips** [comment]
@thockin I think @brendandburns was asking if you think it is OK to set expectations while dropping the carrot and stick. 

Can you explain a bit more about "structured backwards"?

**Comment by thockin** [comment]
"I expect you to do X" implies "or else ...", even if we don't say it.
That is less effective than "if you do X, good-thing Y".  Positive
reinforcement is a stronger motivator than negative reinforcement, and it
lasts longer.  Building up an identity around "I'm a {kontributor,
kollaborator, kommitter} is stronger yet.

Obviously, people are complex animals, but it's well understood that
intrinsic motivators and positive reinforcement are strong.  I don't want
people to feel the implied "or else".

**Comment by philips** [comment]
On Thu, Sep 15, 2016 at 9:38 AM Tim Hockin <notifications@github.com> wrote:

> "I expect you to do X" implies "or else ...", even if we don't say it.
> That is less effective than "if you do X, good-thing Y". Positive
> reinforcement is a stronger motivator than negative reinforcement, and it
> lasts longer. Building up an identity around "I'm a {kontributor,
> kollaborator, kommitter} is stronger yet.
>
> Obviously, people are complex animals, but it's well understood that
> intrinsic motivators and positive reinforcement are strong. I don't want
> people to feel the implied "or else".
>

Got it. Can we simply say something loose like: "If you do code reviews it
will help ensure Kubernetes continues to make solid headline worthy
releases. A track record of code reviews is also an important measure used
when considering adding people as subsystem approvers."

I really just want to try and get this new reviewer system in place before
trying to get an explicit carrot into place. We don't even know if a carrot
is necessary but we do know there are smart engaged folks who's eyes could
be pointed at relevant PRs to review.

Brandon

**Comment by thockin** [comment]
If smart, engaged, willing folks are waiting for a doc to tell then they
are expected to look at code reviews, I don't know what to say.  As
written, this reads to me as a "chore sheet" - something that we need to
point the less-engaged, less-willing people towards.

@sarahnovotny what is between where-we-are and a proper contributor ladder
definition?  If we're going to quantify, let's quantify in a way that
promotes people for good work rather than assigns them work in order to be
promoted.

**Comment by brendandburns** [comment]
@thockin, at the end of the day I'm looking for two things:
   * A page I can point people at when they say "Why did you opt me into this reviewers thingy"
   * A set of numbers that we can plug into automation to decide who is the reviewer (@phillips et al have already built this automation)

**Comment by thockin** [comment]
If we're opting people into reviews, it's going to end in tears.

**Comment by eparis** [comment]
opting people in has 2 aspects.

1. it will make people who may not feel entitled to do reviews feel a sense of ownership and appropriateness that they may not grant themselves. Not everyone has as high of an opinion of themselves as I do and not everyone feels they should be reviewing others and telling others what they did wrong. Opting people in grants them permission and responsibility.
2. opting people in also means that we get me people like me who are terrible reviewers. I'd love to see the accept/reject review thing which was discussed at one point.

Overall I think the tradeoff between empowering/encouraging our community to be leaders and the potential lack of response will be better if we both opt people in and then opt people out who aren't doing such a great job following through. We can't force reviews on people and expect it to always be a positive, but we definitely need a way to prod people up the kubernetes leadership ladder.

I fall on the side of 'opt people in' but hope we track how it goes and make a commitment to iterate.

**Comment by philips** [comment]
Right, @brendandburns we have to have something to write in the email for the potential reviewers and some metric to finalize the selection process. If we get it wrong it is OK to iterate from here; most of the targeted people will already be used to us and probably just opt-in without reading :-P

Brendan, can you drop the carrot/stick and fix up the nits and push?

**Comment by philips** [comment]
@thockin no one is being surprised opted-in. We will email them (see https://github.com/kubernetes/kubernetes/pull/31752) and based on the current metrics I think about ~100 people will be emailed.

**Comment by thockin** [comment]
100 new reviewers seems implausible to me - it suggests the metrics are too
low.

I don't want to block progress, I understand exactly how much we need more
first-line reviewers.  I am just trying wear the hat of a casual
contributor, and being emailed "we expect you to do code reviews" would be
weird, to me.  It feels like it is 180 degrees from what I would expect - I
should be offering to do reviews (or just doing them), rather than being
told to.  What we are missing is some extrinsic motivators.

I will STFU on this, and let it play out if the smart people herein all
feel I am wrong.

**Comment by davidopp** [comment]
I didn't read the whole thread yet but I agree with Tim insofar as opting people in to do reviews seems like it will have a number of negative consequences. Among them, it seems like it will promote quantity over quality, leaving it to the "owners" to do the real reviews.

There must be less draconian ways to prevent "tragedy of the commons." For example, since the real bottleneck is likely to be owners/committers, maybe we could say your PRs will get expedited attention from owner/committers if you are a solid reviewer. This is just an example; I'm sure there are better ideas.

**Comment by philips** [comment]
On Sat, Sep 17, 2016 at 11:44 AM Tim Hockin <notifications@github.com>
wrote:

> 100 new reviewers seems implausible to me - it suggests the metrics are too
> low.
>

Metrics are too low? I don't understand.


> I don't want to block progress, I understand exactly how much we need more
> first-line reviewers. I am just trying wear the hat of a casual
> contributor, and being emailed "we expect you to do code reviews" would be
> weird, to me. It feels like it is 180 degrees from what I would expect - I
> should be offering to do reviews (or just doing them), rather than being
> told to. What we are missing is some extrinsic motivators.
>

I feel this PR, with the removal of the carrot/stick at the end, is a good
starting spot to set expectations, outline the reviewer selection criteria,
and document the reviewer process. @brendandburns can you please remove
this carrot/stick and force push?

The email that needs to go out essentially looks like this:

- We need help doing reviews.
- We know in the past finding work to review has been difficult so we built
automation.
- You were selected, based on previous work, to help review these
subsystems.
- If you opt-in a bot is going to ping you to do reviews on these subsystems

Can @thockin or @davidopp come up with a proposal document that is better
at expressing these bullet points?

Having more people empowered with an explicit task list of things they
should be doing in the community is useful, I think. We are a large
organization now and large organizations generally need to move from
implicit to explicit expectations.

**Comment by thockin** [comment]
On Sep 17, 2016 2:40 PM, "Brandon Philips" <notifications@github.com> wrote:
>
> On Sat, Sep 17, 2016 at 11:44 AM Tim Hockin <notifications@github.com>
> wrote:
>
> > 100 new reviewers seems implausible to me - it suggests the metrics are
too
> > low.
> >
>
> Metrics are too low? I don't understand.

The numbers used.  The net is too wide and catches way more people than is
reasonable.  I think the initial cut of this should catch a dozen-ish
people who would make good reviewers.

>
> > I don't want to block progress, I understand exactly how much we need
more
> > first-line reviewers. I am just trying wear the hat of a casual
> > contributor, and being emailed "we expect you to do code reviews" would
be
> > weird, to me. It feels like it is 180 degrees from what I would expect
- I
> > should be offering to do reviews (or just doing them), rather than being
> > told to. What we are missing is some extrinsic motivators.
> >
>
> I feel this PR, with the removal of the carrot/stick at the end, is a good
> starting spot to set expectations, outline the reviewer selection
criteria,
> and document the reviewer process. @brendandburns can you please remove
> this carrot/stick and force push?
>
> The email that needs to go out essentially looks like this:
>
> - We need help doing reviews.
> - We know in the past finding work to review has been difficult so we
built
> automation.
> - You were selected, based on previous work, to help review these
> subsystems.
> - If you opt-in a bot is going to ping you to do reviews on these
subsystems

I am ok with this, but not to 100 people.

> Can @thockin or @davidopp come up with a proposal document that is better
> at expressing these bullet points?
>
> Having more people empowered with an explicit task list of things they
> should be doing in the community is useful, I think. We are a large
> organization now and large organizations generally need to move from
> implicit to explicit expectations.

ACK people are much more complex than software.

**Comment by brendandburns** [comment]
I'll update this PR today to remove carrot/stick.

@thockin do you want there to be fewer than 100 active reviewers?  I think if we go < 100 we're going to stick with the status quo (~50 Google, ~30 Red Hat, ~20 community)

--brendan

**Comment by sarahnovotny** [comment]
I've generally considered more reviewers as a way to grow trust.  not
immediately presuming that each new review/reviewer is useful, but over
time with secondary (more experienced) reviewers and approvers commenting,
the newly added reviewers learn and the existing approvers see who has the
chops to just get LGTM'd based on prior track record of (extensive and
good) reviews.

sarah

File: community-expectations.md
Comment (jbeda) [inline community-expectations.md]: This is a bit of a run-on sentence.  "... systems.  A second, equally important, goal is the..."

File: community-expectations.md
Comment (jbeda) [inline community-expectations.md]: Agree -- perhaps invite community members to open PRs to change this document as the expectations in our community evolve.

File: community-expectations.md
Comment (jbeda) [inline community-expectations.md]: Another ideas -- some sort of "token" or "credit" to get your PR moved up in the queue based on your contribution to the community.  If you are both submitting and reviewing lots of PRs (not just code but other community work) then you get priority.

However, it is hard to do this without thinking about building automated systems to track it and there will always be room to game the system (small PRs for the sake of a PR, splitting a PR into multiple for credit).  Not sure how far down the path of quantification we want to go.

File: community-expectations.md
Comment (jbeda) [inline community-expectations.md]: I'd love to find a way to help reviewers know when they are falling behind. I've been failing to deal with the deluge of github mail and have been missing issues that have been assigned to me or issues that I've been mentioned in.

**Comment by jbeda** [comment]
I'm a bit late to the party here (sorry!) but how about this.  I apologize if this is well tread ground.

* We define Kubernetes ranks.  
* Those ranks are both based on actions *and* responsibilities.
* People need to opt in to a rank.
* Ranks come with privileges.

Above a certain rank you'll have an expectation to do so many reviews and other work.  You'll also get certain privileges -- the ability to prioritize your own PRs, swag, recognition, fame, guaranteed invites to dev events, etc.

We can make a rank something that folks can put on their resume so that they can demonstrate they are an active real member of the community.

**Comment by thockin** [comment]
That's the contributor ladder that Sarah is working on.  There's some
urgency (??) to get this step in first...

**Comment by thockin** [comment]
My point was that if this catches 100 NEW reviewers, we're doing it wrong.
If that 100 is all the people already doing reviews plus 20 community
folks, OK.

File: community-expectations.md
Comment (brendandburns) [inline community-expectations.md]: done.

File: community-expectations.md
Comment (brendandburns) [inline community-expectations.md]: done.

File: community-expectations.md
Comment (brendandburns) [inline community-expectations.md]: I generally like the idea of special "silver bullets" or some sort of reward for positive reinforcement.

File: community-expectations.md
Comment (brendandburns) [inline community-expectations.md]: done.

File: community-expectations.md
Comment (brendandburns) [inline community-expectations.md]: done.

**Comment by brendandburns** [comment]
@thockin agree, the original PR added ~150 more reviewers which triggered my desire to write this doc. ;)

**Comment by thockin** [comment]
I guess my lingering concern is just the threshold tuning, then.  Do we
have numbers/graphs on the parameters?

**Comment by brendandburns** [comment]
@thockin @davidopp @philips @jbeda @sarahnovotny @smarterclayton 

Comments addressed, carrots and sticks removed (I think)

Please take another look.

Thanks
--brendan

**Comment by brendandburns** [comment]
pretty sure @philips can get us parameters.  Brandon, do you want to tune the parameters for adding ~20 additional reviewers who are not currently active?  (active reviewer defined as having been assigned a review once in the last month?)

Thanks!

**Comment by bgrant0607** [comment]
Sorry, I was OOO. If this is still open in 2 days, I'll try to get to it then.

File: community-expectations.md
Comment (bgrant0607) [inline community-expectations.md]: FYI, we're working on a PR workflow that defines states.
https://docs.google.com/document/d/194MGgJHlm4ovXtvzvkjzzBZG_v47d95nsjuuaUe1Log/edit?ts=57929ed1#heading=h.h8u91v47whwa

File: community-expectations.md
Comment (bgrant0607) [inline community-expectations.md]: Could this doc please go into docs/devel?

Comment (brendandburns) [inline community-expectations.md]: I'm going to wait until I get LGTM on the doc in order to preserve comment history, then I'll move it...

File: community-expectations.md
Comment (thockin) [inline community-expectations.md]: nit: you spelled "two" here and used numeric "3" below

File: community-expectations.md
Comment (thockin) [inline community-expectations.md]: "expected to be available" just sounds so harsh.  How about "We ask that *active contributors* to the project participate in the code review process in areas where that contributor has expertise."

Too wimpy?

File: community-expectations.md
Comment (thockin) [inline community-expectations.md]: This is the one I feel will cast too wide a net.  I don't want to discourage commenting...

**Review by thockin — COMMENTED** [review]
I'll go along with it.  Just some nits.

**Comment by davidopp** [comment]
I'm still not thrilled that this is a separate doc, given that we already have a bazillion docs in docs/devel/ on "how to contribute to Kubernetes." But a bazillion plus one is not much worse than a bazillion, so I guess it doesn't matter.

File: community-expectations.md
Comment (davidopp) [inline community-expectations.md]: s/an//

File: community-expectations.md
Comment (davidopp) [inline community-expectations.md]: I like Tim's rewording

File: community-expectations.md
Comment (davidopp) [inline community-expectations.md]: I think these are all too specific. They're just begging for unintended consequences IMO. But I agree this one is the most problematic.

File: community-expectations.md
Comment (davidopp) [inline community-expectations.md]: I didn't understand this one. What does it mean for someone who is not a reviewer to mark something as LGTM?

File: community-expectations.md
Comment (davidopp) [inline community-expectations.md]: @mml wrote a cool script for this, and I have seen at least one other, though I can't remember where.

**Review by davidopp — CHANGES_REQUESTED** [review]

File: community-expectations.md
Comment (brendandburns) [inline community-expectations.md]: Maybe I'll make this "Commented on more than 5 PRs?"  I think if you are actively commenting on PRs then you are by definition an active reviewer and should be considered for the pool.

Also there are several people in the project who do a lot of reviewing and commenting, but often don't actually write any PRs.

**Comment by brendandburns** [comment]
I'll address comments today and push a (hopefully) final draft.

@davidopp if you think there is a better home, let me know and I'll move it there.

I personally think this is a more important document than most because it shapes the shape of the community, but that's my thought.

File: community-expectations.md
Comment (bgrant0607) [inline community-expectations.md]: Rather than "previous year", how about something like "trailing 12 months", to resolve confusion about whether today this would mean in 2015 or Sept 2015 to Sept 2016.

File: community-expectations.md
Comment (bgrant0607) [inline community-expectations.md]: It's possible to mark PRs as LGTM without otherwise commenting.

File: community-expectations.md
Comment (bgrant0607) [inline community-expectations.md]: I like Tim's wording, also, because it sounds more active ("participate").

File: community-expectations.md
Comment (bgrant0607) [inline community-expectations.md]: What are "contributor" permissions? Write or admin access?

File: community-expectations.md
Comment (bgrant0607) [inline community-expectations.md]: I plan to automatically reassign such PRs and to track reviewer stats.

**Comment by bgrant0607** [comment]
@k8s-bot test this issue: #33481

**Comment by bgrant0607** [comment]
I made some comments, but would like to merge this and iterate. If more changes are made, great. If not, I'll LGTM anyway later today.

File: community-expectations.md
Comment (brendandburns) [inline community-expectations.md]: done.

File: community-expectations.md
Comment (brendandburns) [inline community-expectations.md]: removed issues, focused on PRs alone.

File: community-expectations.md
Comment (brendandburns) [inline community-expectations.md]: Also, if you LGTM one thing, that makes you active, even if that's your only comment on a PR in that month.

File: community-expectations.md
Comment (brendandburns) [inline community-expectations.md]: done.

File: community-expectations.md
Comment (brendandburns) [inline community-expectations.md]: done.

File: community-expectations.md
Comment (brendandburns) [inline community-expectations.md]: moved to "previous 12 months"

File: community-expectations.md
Comment (brendandburns) [inline community-expectations.md]: ah, 'collaborator' is the github term: https://help.github.com/articles/permission-levels-for-a-user-account-repository/

switched to that.

File: community-expectations.md
Comment (brendandburns) [inline community-expectations.md]: Yeah, I think this is what we want in the limit.

File: community-expectations.md
Comment (brendandburns) [inline community-expectations.md]: moved.

**Comment by brendandburns** [comment]
Ok, I addressed all of the comments, and squashed, this is ready for merge as far as I'm concerned.

thanks all for the feedback, please let me know what you'd like to see in further revisions.

**Comment by bgrant0607** [comment]
Verification checks failed. Please run hack/update-munge-docs.sh

**Comment by bgrant0607** [comment]
LGTM otherwise. Feel free to re-apply after you push fixes.

**Comment by brendandburns** [comment]
update-munge-docs run, re-applied 'lgtm'

**Comment by philips** [comment]
LGTM. I wish there was some way to provide a link to all PRs that are active for a reviewer but that will require more tooling. Ship it!

**Pull request merged**
